src/Compilers/Core/Portable/Collections/DictionaryExtensions.cs

```diff
@@ -31,5 +32,21 @@ internal static class DictionaryExtensions
                 return value;
             }
         }
+
+        public static bool TryAdd<T>(
```

Comment:
Consider just making this a conditional extension method on `ImmutableSegmentedDictionary<>.Builder`, rather than on `IDictionary`. If it's on `IDictionary`, it'll appear for things it shouldn't, like `ImmutableDictionary`, where it's a dangerous nop.

Reply:
➡️ Now implemented in `ImmutableSegmentedDictionaryBuilderExtensions`